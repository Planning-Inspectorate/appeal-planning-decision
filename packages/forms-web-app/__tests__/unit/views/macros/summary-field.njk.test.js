const nunjucksTestRenderer = require('../nunjucks-render-helper');

describe('views/macros/summary-field', () => {
  it('should render without a value or attributes', () => {
    expect(
      nunjucksTestRenderer.renderString(`
    {%- from 'macros/summary-field.njk' import summaryField with context -%}
    {{- summaryField() -}}
    `)
    ).toMatchSnapshot();
  });

  it('should render with a value and without attributes', () => {
    expect(
      nunjucksTestRenderer.renderString(`
    {%- from 'macros/summary-field.njk' import summaryField with context -%}
    {{- summaryField(value = 'an example value' ) -}}
    `)
    ).toMatchSnapshot();
  });

  it('should render without a value and with attributes', () => {
    expect(
      nunjucksTestRenderer.renderString(`
    {%- from 'macros/summary-field.njk' import summaryField with context -%}
    {{- summaryField(attributes = { 'a': 'b', 'c': 'd' } ) -}}
    `)
    ).toMatchSnapshot();
  });

  it('should render with a value and attributes', () => {
    expect(
      nunjucksTestRenderer.renderString(`
    {%- from 'macros/summary-field.njk' import summaryField with context -%}
    {{- summaryField(value = 'an example value', attributes = { 'a': 'b', 'c': 'd' } ) -}}
    `)
    ).toMatchSnapshot();
  });
});
