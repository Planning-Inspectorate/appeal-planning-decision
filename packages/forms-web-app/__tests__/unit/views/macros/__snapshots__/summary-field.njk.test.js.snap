// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`views/macros/summary-field should render with a value and attributes 1`] = `
"
  <span a=\\"b\\" c=\\"d\\">an example value</span>
"
`;

exports[`views/macros/summary-field should render with a value and without attributes 1`] = `
"
  <span>an example value</span>
"
`;

exports[`views/macros/summary-field should render without a value and with attributes 1`] = `
"
  <span a=\\"b\\" c=\\"d\\"></span>
"
`;

exports[`views/macros/summary-field should render without a value or attributes 1`] = `
"
  <span></span>
"
`;
